### DISCUSSION TITLE
What are encoder plugins used for?

### DISCUSSION THREAD

**Discussion by doronbehar** [MusicPlayerDaemon/MPD#1267] [Q&A] [answered]
The [documentation](https://mpd.readthedocs.io/en/stable/plugins.html#encoder-plugins) doesn't explain much, I mean: What do they encode? Are they sort of a replacement to resampler plugins?

**Comment by MaxKellermann** [comment, marked as answer]
Several output plugins need an encoder, e.g. the "recorder" plugin which records all played songs to hard disk; the "shout" plugin encodes to Vorbis or MP3 before streaming to Icecast; the "httpd" plugin encodes to whatever you want for streaming to arbitrary HTTP clients.
For normal, local playback, you don't need any encoder.
